extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java

```diff
@@ -0,0 +1,385 @@
+        items.add(new ReflectiveClassBuildItem(true, true, RemoveFileExtension.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Repeat.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ReplaceAll.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ReplaceFirst.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Round.class));
+        items.add(new ReflectiveClassBuildItem(true, true, SeparateByDash.class));
+        items.add(new ReflectiveClassBuildItem(true, true, SeparateByUnderscore.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Split.class));
+        items.add(new ReflectiveClassBuildItem(true, true, StartsWith.class));
+        items.add(new ReflectiveClassBuildItem(true, true, StringList.class));
+        items.add(new ReflectiveClassBuildItem(true, true, SubString.class));
+        items.add(new ReflectiveClassBuildItem(true, true, SubStringAfter.class));
+        items.add(new ReflectiveClassBuildItem(true, true, SubStringBefore.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Subtract.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Trim.class));
+        items.add(new ReflectiveClassBuildItem(true, true, TrimLeft.class));
+        items.add(new ReflectiveClassBuildItem(true, true, TrimRight.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Uppercase.class));
+        items.add(new ReflectiveClassBuildItem(true, true, UppercaseChar.class));
+        items.add(new ReflectiveClassBuildItem(false, true, true, VolumeUnitType.class));
+        items.add(new ReflectiveClassBuildItem(true, false, XmlModule.class));
+        items.add(new ReflectiveClassBuildItem(false, true, false, DateFieldActions.class));
+        items.add(new ReflectiveClassBuildItem(false, true, true, NodeType.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Restriction.class));
+        items.add(new ReflectiveClassBuildItem(false, true, true, RestrictionType.class));
+        items.add(new ReflectiveClassBuildItem(true, true, Restrictions.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlDataSource.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlField.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlNamespace.class));
+        items.add(new ReflectiveClassBuildItem(true, true, XmlNamespaces.class));
+        items.add(new ReflectiveClassBuildItem(true, true, CsvDataSource.class));
+        items.add(new ReflectiveClassBuildItem(true, true, CsvField.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ADMDigest.class));
+        items.add(new ReflectiveClassBuildItem(true, true, DataSourceMetadata.class));
+        items.add(new ReflectiveClassBuildItem(false, true, true, InspectionType.class));
+        items.add(new ReflectiveClassBuildItem(true, true, ValueContainer.class));
+        return items;
```

Comment:
Maybe as follow up we may use jandex to discover the types we need to register for reflection, best would be if atlasmap weere providing the jamdex index but if not, we can explicitly index atlasmap jars

Reply:
@lburgazzoli : I've created an issue to follow up  with this one : [2241](https://github.com/apache/camel-quarkus/issues/2241)